webpack/assets/javascripts/react_app/components/MemoryAllocationInput/MemoryAllocationInput.js

```diff
@@ -1,121 +1,81 @@
-    setValue(defaultValue);
-  }, [defaultValue]);
-
-  useEffect(() => {
-    if (value > maxValue && maxValue !== undefined) {
-      setValidationState('error');
-    } else if (
-      value > recommendedMaxValue &&
-      recommendedMaxValue !== undefined
-    ) {
-      setValidationState('warning');
+    const valueBytes = valueMB * MEGABYTES;
+    if (maxValue && valueBytes > maxValue) {
+      setWarning(null);
+      setError(
+        sprintf(
+          __('Specified value is higher than maximum value %s'),
+          `${maxValue / MEGABYTES} ${MB_FORMAT}`
+        )
+      );
+    } else if (recommendedMaxValue && valueBytes > recommendedMaxValue) {
+      setError(null);
+      setWarning(
+        sprintf(
+          __('Specified value is higher than recommended maximum %s'),
+          `${recommendedMaxValue / MEGABYTES} ${MB_FORMAT}`
+        )
+      );
     } else {
-      setValidationState(undefined);
+      setWarning(null);
     }
-  }, [recommendedMaxValue, maxValue, value]);
-
-  const handleValueIncrease = () => {
-    setValue(value * 2);
-  };
-
-  const handleValueDecrease = () => {
-    setValue(value / 2);
-  };
-
-  const handleTypedValue = v => {
-    setValue(v);
-  };
+  }, [valueMB, recommendedMaxValue, maxValue, setError, setWarning]);
 
   const handleChange = v => {
-    if (v === value + 1) {
-      handleValueIncrease();
-    } else if (v === value - 1) {
-      handleValueDecrease();
-    } else {
-      handleTypedValue(v);
-    }
-    onChange(value);
-  };
-
-  const format = v => `${v} ${MB_FORMAT}`;
-
-  const helpBlock = () => {
-    if (validationState === 'warning') {
-      return (
-        <HelpBlock>
-          <WarningTriangleIcon className="warning-icon" />
-          {__('Specified value is higher than recommended maximum')}
-        </HelpBlock>
-      );
-    } else if (validationState === 'error') {
-      return (
-        <HelpBlock>
-          <ErrorCircleOIcon className="error-icon" />
-          {__('Specified value is higher than maximum value')}
-        </HelpBlock>
-      );
+    if (v === valueMB + 1) {
+      v = valueMB * 2;
+    } else if (v === valueMB - 1) {
+      v = Math.floor(valueMB / 2);
     }
-    return undefined;
+    setValueMB(v);
+    onChange(v * MEGABYTES);
   };
 
-  const parser = str => str.replace(/\D/g, '');
   return (
-    <Grid>
-      <Row>
-        <Col>
-          <NumericInput
-            value={value}
-            id={id}
-            format={format}
-            parser={parser}
-            onChange={handleChange}
-            label={label}
-            disabled={disabled}
-            minValue={minValue}
-          />
-          <input type="hidden" name={name} value={value * MEGABYTES} />
-        </Col>
-      </Row>
-      <Row>
-        <Col md={2} />
-        <Col md={4} className="form-group">
-          {validationState !== undefined && helpBlock()}
-        </Col>
-      </Row>
-    </Grid>
+    <>
+      <RCInputNumber
```

Comment:
Uses the external component directly, as the FormInput component takes care of the labels and other stuff.